src/platform/unix/mod.rs

```diff
@@ -873,50 +924,37 @@ fn recv(fd: c_int, blocking_mode: BlockingMode)
     Ok((main_data_buffer, channels, shared_memory_regions))
 }
 
-#[cfg(target_os="android")]
-fn maybe_unlink(_: *const c_char) -> c_int {
-    // Calling `unlink` on a file stored on an sdcard immediately deletes it.
-    // FIXME: use a better temporary directory than the sdcard via the Java APIs
-    // and threading that value into Servo.
-    // https://code.google.com/p/android/issues/detail?id=19017
-    0
+unsafe fn create_filename() -> CString {
+    let mut tmp_string = TEMP_FILE_TEMPLATE.to_owned();
+    let addition: String = thread_rng().gen_ascii_chars().take(6).collect();
+    tmp_string.push_str(&addition);
+    CString::new(tmp_string).unwrap()
 }
 
-#[cfg(not(target_os="android"))]
-unsafe fn maybe_unlink(c: *const c_char) -> c_int {
-    libc::unlink(c)
+#[cfg(any(feature="force-shm", target_os="freebsd"))]
+#[inline]
+unsafe fn create_shmem(name: *const c_char, length: usize) -> c_int {
+    let fd = libc::shm_open(name, libc::O_CREAT | libc::O_RDWR, 0660);
```

Comment:
Why `660`, not `600`?

(Though in fact it probably should be `000`, since we don't ever open the same object again anyway -- we just clone the existing FD, which is already open with `O_RDWR`...)

Also, `0660` doesn't do what you think it does :-) In Rust, you'd need `0o660`.

Last but not least, it's probably good to add `O_EXCL` just to be safe.